homeassistant/components/tesla_fleet/coordinator.py

```diff
@@ -50,12 +51,53 @@ def flatten(data: dict[str, Any], parent: str | None = None) -> dict[str, Any]:
     return result
 
 
+class RateCalculator:
+    """Calculate the consumption and remaining rate of a rate limit."""
+
+    def __init__(self, limit, period=86400, min_wait=0, max_wait=None) -> None:
```

Comment:
2 things: would this be something for the library? And please type

Reply:
Probably, I did originally implement it in the library, so I can pull it back out and use a bump instead.

Reply:
I wasn't sure where it should end up. It is possible this integration will be the only place it gets used as I am likely going to take a different server-side approach with Teslemetry, and Tessie has no rate limit concerns.

Reply:
Aah check